### PR TITLE
Tutoring schedule algorithm

### PR DESCRIPTION
**Pull request by vikranth22446**
Fixes issue #54

### CODE REVIEW

File: hknweb/tutoring/models.py [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
     id = models.AutoField(primary_key=True)
     name = models.CharField(max_length=255, null=False)
 
 
+class Tutor(models.Model):
+    id = models.AutoField(primary_key=True)
+    name = models.CharField(max_length=255)
+    courses = models.ManyToManyField(Course)
+    adjacency = models.IntegerField()
+    user = models.OneToOneField(User, on_delete=DO_NOTHING, related_name="tutor")
```
Comment (jameslzhu) [inline hknweb/tutoring/models.py]: on_delete should probably be set to CASCADE.

File: hknweb/tutoring/models.py [hunk 2]
```diff
@@ -56,9 +70,72 @@ def start_time(self):
...
+    ]
     id = models.AutoField(primary_key=True)
-    name = models.CharField(max_length=255)
-    slots = models.ManyToManyField(Slot)
-    courses = models.ManyToManyField(Course)
+    courses = models.ForeignKey(Course, on_delete=DO_NOTHING)
+    level = models.IntegerField(choices=COURSE_PERFS)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    tutor = models.OneToOneField(Tutor, on_delete=DO_NOTHING, related_name="coursepreference")
```
Comment (jameslzhu) [inline hknweb/tutoring/models.py]: ditto.

File: hknweb/tutoring/models.py [hunk 3]
```diff
@@ -56,9 +70,72 @@ def start_time(self):
...
+    available = 2
+    unavailable = 3
+
+    PREFERENCES_CHOICES = [
+        (preferred, 'Preferred'),
+        (available, 'Available'),
+        (unavailable, 'Not Available')
+    ]
+    id = models.AutoField(primary_key=True)
+    tutor = models.OneToOneField(Tutor, on_delete=DO_NOTHING, related_name="availabilities")
```
Comment (jameslzhu) [inline hknweb/tutoring/models.py]: ditto.

File: hknweb/tutoring/models.py [hunk 4]
```diff
@@ -56,9 +70,72 @@ def start_time(self):
...
+    COURSE_PERFS = [
+        (current, 'Current'),
+        (completed, 'Completed'),
+        (preferred, 'Preferred')
+    ]
     id = models.AutoField(primary_key=True)
-    name = models.CharField(max_length=255)
-    slots = models.ManyToManyField(Slot)
-    courses = models.ManyToManyField(Course)
+    courses = models.ForeignKey(Course, on_delete=DO_NOTHING)
```
Comment (jameslzhu) [inline hknweb/tutoring/models.py]: ditto. If the course gets deleted, we probably shouldn't have preferences to a non-existent course.

**Comment by bri25yu** [comment]
Stale, updated work found at a new PR https://github.com/compserv/hknweb/pull/254

**Pull request closed without merging**